File: FilmHive/filmHive.Model/List.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace filmHive.Model
{
    public class List
    {
        public int ListId { get; set; }
        public int UserId { get; set; }
        public string ListName { get; set; } = null!;
        public string? ListDescription { get; set; }
        public bool IsDeleted { get; set; }
        public bool IsActive { get; set; }
        public DateTime? TimeOfDeletion { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? ModifiedAt { get; set; }
        public int? ModifiedBy { get; set; }
        public virtual ICollection<ListFilm> ListFilms { get; set; } = new List<ListFilm>();
        public virtual User? User { get; set; } 
    }
}

File: FilmHive/UI/filmHive/models/list.js
import User from "./user";

class List{
    constructor(data) {
        this.listId = data.listId;
        this.userId = data.userId;
        this.listName = data.listName;
        this.listDescription = data.listDescription;
        this.isDeleted = data.isDeleted;
        this.isActive = data.isActive;
        this.timeOfDeletion = data.timeOfDeletion; 
        this.createdAt = data.createdAt;
        this.modifiedAt = data.modifiedAt;
        this.modifiedBy = data.modifiedBy;
        // this.listFilms = data.listFilms || [];
        this.user = data.user ? new User(data.user) : null;
    }
}
export default List;